src/backend/src/services/projects.services.ts

```diff
@@ -506,4 +507,59 @@ export default class ProjectsService {
 
     return projectTransformer(deletedProject);
   }
+
+  /**
+   * Toggles a user's favorite status on a projects
+   * @param projectId the project id to be favorited/unfavorited
+   * @param user the user who is changing the role
+   * @param favorite boolean representing favoriting/unfavoriting
+   * @returns the project that the user has favorited/unfavorited
+   * @throws if the project id doesn't exist
+   */
+  static async toggleFavorite(wbsNumber: WbsNumber, user: PrismaUser): Promise<Project> {
+    if (!isProject(wbsNumber)) throw new HttpException(400, `${wbsPipe(wbsNumber)} is not a valid project WBS #!`);
+    const { carNumber, projectNumber, workPackageNumber } = wbsNumber;
+
+    const projectInfo = await prisma.project.findFirst({
+      where: {
+        wbsElement: {
+          carNumber,
+          projectNumber,
+          workPackageNumber
+        }
+      },
+      include: {
+        ...projectQueryArgs.include,
+        favoritedBy: true
+      }
+    });
+
+    if (!projectInfo) throw new NotFoundException('WBS Element', wbsPipe(wbsNumber));
+    const favorited = projectInfo.favoritedBy.filter((allUsers) => allUsers.userId === user.userId).length;
+
+    favorited
+      ? await prisma.user.update({
+          where: { userId: user.userId },
+          data: {
+            favoriteProjects: {
+              disconnect: {
+                projectId: projectInfo.projectId
+              }
+            }
+          }
+        })
+      : await prisma.user.update({
+          where: { userId: user.userId },
+          data: {
+            favoriteProjects: {
+              connect: {
+                projectId: projectInfo.projectId
+              }
+            }
+          }
+        });
+
+    const { favoritedBy: _, ...project } = projectInfo;
```

Comment:
id just add favorited by to project query args but anthony might disagree then ud be able to get rid of that weird include statement at the top.

Reply:
fair enough... this line would still be necessary though

Reply:
no it wouldnt becasue project transformer takes in project query args

Reply:
ooh okay